apps/mini-app/src/components/GeoCoder/GeoCoder.jsx

```diff
@@ -0,0 +1,39 @@
+import { useEffect } from 'react';
+import { marker, Control, Marker } from 'leaflet';
+import { useMap } from 'react-leaflet';
+import icon from './MarkerIcon';
+import 'leaflet-control-geocoder/dist/Control.Geocoder.css';
+import 'leaflet-control-geocoder/dist/Control.Geocoder.js';
+
+export function GeoCoder() {
+  const map = useMap();
+
+  useEffect(() => {
+    const geocoder = Control.Geocoder.nominatim();
+
+    Control.geocoder({
+      position: 'topright',
+      placeholder: 'Search...',
+      defaultMarkGeocode: false,
+      query: '',
+      geocoder,
+    })
+    .on('markgeocode', (e) => {
```

Comment:
I think we need to remove the event handler on unmount

Reply:
Also remove the marker on unmount

Reply:
Done